File: Code Examples/Heavy/HeavyCorrectExample.cs
//should not timeout (O(n*log n))
class HeavyCorrectExample
{
    public class Solution
    {
        public double FractionalKnapsack(Item[] items, int capacity)
        {
            Array.Sort(items, (a, b) => b.Ratio.CompareTo(a.Ratio));

            double totalValue = 0.0;

            foreach (var item in items)
            {
                if (capacity >= item.Weight)
                {
                    capacity -= item.Weight;
                    totalValue += item.Value;
                }
                else
                {
                    totalValue += item.Ratio * capacity;
                    break;
                }
            }

            return totalValue;
        }
    }

}

File: Code Examples/Heavy/HeavyCorrectExample.js
class Solution {
    fractionalKnapsack(items, capacity) {
        items.sort((a, b) => b.ratio - a.ratio);

        let totalValue = 0.0;

        for (const item of items) {
            if (capacity >= item.weight) {
                capacity -= item.weight;
                totalValue += item.value;
            } else {
                totalValue += item.ratio * capacity;
                break;
            }
        }

        return totalValue;
    }
}